app/Models/Extensions/AlbumGetters.php

```diff
@@ -68,24 +68,30 @@ public function get_all_photos()
 			->where('albums._rgt', '<=', $this->_rgt);
 	}
 
-	public function get_thumbs()
+	public function get_thumb(): ?Thumb
 	{
-		[$sort_col, $sort_order] = $this->get_sort();
+		if ($this->cover != null) {
+			$cover = $this->cover;
+		} else {
+			[$sort_col, $sort_order] = $this->get_sort();
+
+			$sql = $this->get_all_photos();
+
+			//? apply safety filter : Do not leak pictures which are not ours
+			$forbiddenID = resolve(PublicIds::class)->getNotAccessible();
```

Comment:
I'm guessing this might be responsible for the unrelated issue I'm seeing with thumbs of smart albums for non-admin users?

Reply:
Indeed.

Reply:
fixed.